src/EnergyPlus/ConvectionCoefficients.cc

```diff
@@ -2412,678 +1673,171 @@ void GetUserConvectionCoefficients(EnergyPlusData &state)
-        }
-    } break;
-    case SurfacesType::AllInteriorFloors: {
-        SurfacesOfType = false;
-        SurfaceCountOutside = 0;
-        SurfaceCountInside = 0;
-        for (SurfNum = 1; SurfNum <= state.dataSurface->TotSurfaces; ++SurfNum) {
-            if (!Surface(SurfNum).HeatTransSurf) continue;
-            if (Surface(SurfNum).ExtBoundCond <= 0) continue; // Exterior surfaces
-            if (Surface(SurfNum).Class != SurfaceClass::Floor) continue;
-            SurfacesOfType = true;
-            if (ConvectionType == "OUTSIDE") {
-                if (Surface(SurfNum).OSCPtr > 0) continue;
-                if (state.dataSurface->SurfExtConvCoeffIndex(SurfNum) != 0) {
-                    if (state.dataGlobal->DisplayExtraWarnings) {
-                        ShowWarningError(state,
-                                         format("User Supplied Convection Coefficients, Multiple Surface Assignments=\"{}\", not overwriting already "
-                                                "assigned value for (Outside) in Surface={}",
-                                                SurfaceTypes,
-                                                Surface(SurfNum).Name));
-                    } else {
-                        ++SurfaceCountOutside;
-                    }
-                } else {
-                    state.dataSurface->SurfExtConvCoeffIndex(SurfNum) = Value;
-                }
-            } else {
-                if (state.dataSurface->SurfIntConvCoeffIndex(SurfNum) != 0) {
-                    if (state.dataGlobal->DisplayExtraWarnings) {
-                        ShowWarningError(state,
-                                         format("User Supplied Convection Coefficients, Multiple Surface Assignments=\"{}\", not overwriting already "
-                                                "assigned value for (Inside) in Surface={}",
-                                                SurfaceTypes,
-                                                Surface(SurfNum).Name));
-                    } else {
-                        ++SurfaceCountInside;
-                    }
-                } else {
-                    state.dataSurface->SurfIntConvCoeffIndex(SurfNum) = Value;
-                }
-            }
-        }
-        if (!state.dataGlobal->DisplayExtraWarnings && (SurfaceCountOutside > 0 || SurfaceCountInside > 0)) {
-            if (SurfaceCountOutside > 0) {
-                OverwriteMessage = format("{} Outside", SurfaceCountOutside);
-            }
-            if (SurfaceCountInside > 0) {
-                OverwriteMessage = format("{} Inside", SurfaceCountInside);
-            }
+    // This subroutine applies a convection type to a set of surfaces.  This is
+    // one of the "regular" convection types and becomes a "negative" convection
+    // type to that surface.
+
+    // SUBROUTINE LOCAL VARIABLE DECLARATIONS:
+    if (state.dataSurface->SurfaceFilterLists[static_cast<int>(surfaceFilter)].size() == 0) {
```

Comment:
The various surface filter lists are now created and saved during a single initial pass.